invoicing/src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -0,0 +1,108 @@
+package pl.futurecollars.invoicing.db.file;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.utils.FilesService;
+import pl.futurecollars.invoicing.utils.IdService;
+import pl.futurecollars.invoicing.utils.JsonService;
+
+public class FileBasedDatabase implements Database {
+
+  private final FilesService filesService;
+  private final JsonService jsonService;
+  private final IdService idService;
+  private final Path databasePath;
+
+  public FileBasedDatabase(Path databasePath, FilesService filesService, JsonService jsonService, IdService idService)
+      throws IOException {
+    this.filesService = filesService;
+    this.jsonService = jsonService;
+    this.idService = idService;
+    this.databasePath = databasePath;
+    filesService.createFile(databasePath.toString());
```

Comment:
To lepiej wynieść do osobnej metody, której nazwa powie nam co robi (biznesowo) ta linijka. Obecnie można się tylko domyślać.

Reply:
Warto też sprawdzać, czy taki plik już istnieje. Jeśli tak to nie ma po co tworzyć nowego.

Reply:
Metoda mogła by się nazywać np. initialize